### DISCUSSION TITLE
Dependency management for a module (to avoid users having to redeclare the same dependencies)

### DISCUSSION THREAD

**Discussion by hlassiege** [nuxt/nuxt#26660] [Questions] [answered]
Hi,

I'm authoring a template that can be used to start a new blog : [bloggrify](https://github.com/hlassiege/bloggrify)

This app has several dependencies, nuxt-image, nuxt-content, fuse etc...

A demo app is visible here: https://github.com/hlassiege/demo-mistral

I use the "extends" mechanism : 

```
    extends: [
        'github:hlassiege/bloggrify#main',
        'github:hlassiege/bloggrify-mistral#main',
    ],
```
However,  I'm a bit unsatisfied by the developer experience because the demo application has to redeclare all the dependencies of the initial module(s), bloggrify (and the theme bloggrify-mistral)

it's not very satisfactory, since the end user shouldn't have to know which dependencies to install or remove if they're no longer needed.

What would be the best approach to avoid this?

I should point out that I'm more of a Java dev than a front-end dev, which explains why I may not be familiar with some of the basic mechanisms in the front-end world :) (In Java, dependencies are transitive, so the problem doesn't exist)

**Comment by hlassiege** [comment, marked as answer]
Ok, @victorgarciaesgi helped me on that ^^

Apparently, if I use 

```
    extends: [
        'github:hlassiege/bloggrify#main',
        'github:hlassiege/bloggrify-mistral#main',
    ],
```
    
it don't work. But if I use an extends to an npm package, it works 
    
```
        extends: [
        @bloggrify/core',
        'github:hlassiege/bloggrify-mistral#main',
    ],
```

**Reply by v0id-4lpz** [reply]
What if we dont want to deal with npm packages ? 😕
